internal/driver/glfw/loop.go

```diff
@@ -67,8 +67,15 @@ func runOnDraw(w *window, f func()) {
 	<-done
 }
 
+// Preallocate to avoid allocations on every drawSingleFrame.
+// Note that the capacity of this slice can only grow,
+// but its length will never be longer than the total number of
+// window canvases that are dirty on a single frame.
+// So its memory impact should be negligible and does not
+// need periodic shrinking.
+var refreshingCanvases = make([]fyne.Canvas, 0)
```

Comment:
You don't have make it into zero size. Doing so does an unnecessary call to the runtime per https://stackoverflow.com/a/29164565.  Just do this instead:
```suggestion
var refreshingCanvases []fyne.Canvas
```

Reply:
The alternative I guess is to use a best-guess size rather than 0 in the initializer, but at this point it really is unnecessary optimization so I'll just go with your suggestion :)

Reply:
To be fair, I do not necessarily only say this because it technically should be faster but I do prefer to only make a slice when I need to set a non default capacity. Not using make for zero size does make me have to parse less code if that makes sense?

Reply:
I see now that it already was make'd to zero. Not you fault but I have never really understood why people do it. It is more code to write as well ;)